src/table/components/head/HeadCellMenu.tsx

```diff
@@ -46,15 +65,89 @@ export default function HeadCellMenu({ column, tabIndex }: HeadCellMenuProps) {
                   embedListbox();
                 },
                 icon: <Search />,
-                isDisabled: false,
+                enabled: true,
+              },
+            ],
+            [
+              {
+                id: 1,
+                itemTitle: translator.get('SNTable.MenuItem.SelectAll'),
+                onClick: async () => {
+                  await fieldInstance?.selectAll();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectAll />,
+                enabled: selectionAnalysis.canSelectAll,
+              },
+              {
+                id: 2,
+                itemTitle: translator.get('SNTable.MenuItem.ClearSelections'),
+                onClick: async () => {
+                  await fieldInstance?.clear();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <ClearSelections />,
+                enabled: selectionAnalysis.clearSelections,
+              },
+              {
+                id: 3,
+                itemTitle: translator.get('SNTable.MenuItem.SelectPossible'),
+                onClick: async () => {
+                  await fieldInstance?.selectPossible();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectPossible />,
+                enabled: selectionAnalysis.canSelectPossible,
+              },
+              {
+                id: 4,
+                itemTitle: translator.get('SNTable.MenuItem.SelectAlternative'),
+                onClick: async () => {
+                  await fieldInstance?.selectAlternative();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectAlternative />,
+                enabled: selectionAnalysis.canSelectAlternative,
+              },
+              {
+                id: 5,
+                itemTitle: translator.get('SNTable.MenuItem.SelectExcluded'),
+                onClick: async () => {
+                  await fieldInstance?.selectExcluded();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectExcluded />,
+                enabled: selectionAnalysis.canSelectExcluded,
               },
             ],
           ]
         : []),
     ],
-    [translator, showSearchMenuItem]
+    [translator, showSearchMenuItem, fieldInstance, selectionAnalysis]
   );
 
+  const checkStateCountByKey = <T,>(keys: (keyof T)[], obj: T): boolean => {
+    return keys.some((key) => obj[key] > 0);
+  };
+
+  const runSelectionAnalysis = (layout: TableLayout) => {
```

Comment:
looks like these only depend on `layout`, so I think they can be put in a util file